src/edu/washington/escience/myria/operator/AbstractDbInsert.java

```diff
@@ -18,6 +18,6 @@
    * @param child the source of tuples.
    */
   public AbstractDbInsert(final Operator child) {
-    super(child, TupleBatch.BATCH_SIZE);
+    super(child, (child != null) ? TupleUtils.getBatchSize(child.generateSchema()) : 1);
```

Comment:
I'm confused by this `coalesce` or `threshold`, I'm putting all the comments here:
- Why do we need it?
- The constructor of `AbstractDbInsert`, when it's called, may not have `child` being set. For example, from JSON. So here you always have 1 even for non-blob types. A better place is `init()` where child is set. 
- Is it only a feature of `DbInsert`? If so, we shouldn't put anything in `RootOperator`.
- The procedure of the actual coalescing (`RootOperator.getCoalesced`) can be implemented by using `TupleBuffer/TupleBatchBuffer`. And actually I think `ConcatColumn` seems to be a dangerous thing since it may go larger than the size of a `TupleBatch` but nothing prevents us from creating a `TupleBatch` with them. Since it's not used anywhere except here, I'd like to remove it later on.
- The doc says "0 or negative means do not coalesce" but there is no such a check. Also we need to be consistent: is the passing criterion `>` or `>=` threshold?